src/components/vm/snapshots/vmSnapshotsCreateModal.jsx

```diff
@@ -58,6 +63,40 @@
     );
 };
 
+function getDefaultMemoryPath(vm, snapName) {
+    // Choosing a default path where memory snapshot should be stored might be tricky. Ideally we want
+    // to store it in the same directory where the primary disk (the disk which is first booted) is stored
+    // If howver no such disk can be found, we should fallback to libvirt's default /var/lib/libvirt
+    const devices = getSortedBootOrderDevices(vm).filter(d => d.bootOrder &&
+                                                              d.device.device === "disk" &&
+                                                              d.device.type === "file" &&
+                                                              d.device.source.file);
+    if (devices.length > 0) {
+        const primaryDiskPath = devices[0].device.source.file;
+        const directory = primaryDiskPath.substring(0, primaryDiskPath.lastIndexOf("/") + 1);
+        return directory + snapName;
+    } else {
+        if (vm.connectionName === LIBVIRT_SYSTEM_CONNECTION)
+            return "/var/lib/libvirt/memory/" + snapName;
+        else if (current_user)
+            return current_user.home + "/.local/share/libvirt/memory/" + snapName;
+    }
+
+    return "";
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1280-20240129-101752-d3606ddd-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)